yt_dlp/extractor/nytimes.py

```diff
@@ -125,138 +91,427 @@ class NYTimesIE(NYTimesBaseIE):
+        for image in thumbs:
+            image_url = image.get('url')
+            if not image_url:
+                continue
+            thumbnails.append({
+                'url': image_url,
+                'width': int_or_none(image.get('width')),
+                'height': int_or_none(image.get('height')),
+            })
+        return thumbnails
 
-        episode_title = track['title']
-        video_url = track['source']
+    def _real_extract(self, url):
+        page_id = self._match_id(url)
+        webpage = self._download_webpage(url, page_id)
+        article_json = self._search_json(
+            r'window\.__preloadedData\s*=', webpage, 'media details', page_id,
+            transform_source=lambda x: x.replace('undefined', 'null'))['initialData']['data']['article']
+        blocks_json = traverse_obj(
+            article_json, ('sprinkledBody', 'content', ..., ('ledeMedia', None), lambda _, v: v['__typename'] in ('Video', 'Audio')))
+
+        art_title = remove_end(self._html_extract_title(webpage), ' - The New York Times')
+        art_description = traverse_obj(
+            article_json, ('sprinkledBody', 'content', ..., 'summary', 'content', ..., 'text'),
+            get_all=False) or self._html_search_meta(['og:description', 'twitter:description'], webpage)
+        art_upload_date = traverse_obj(article_json, ('firstPublished'))
+        creator = ', '.join(
+            traverse_obj(article_json, ('bylines', ..., 'creators', ..., 'displayName'), get_all=True))
+
+        # more than 1 video in the article, treat it as a playlist
+        if len(blocks_json) > 1:
+            entries = []
+            for block_json in blocks_json:
+                content = self._extract_content_from_block(block_json)
+                thumbnails = self._extract_thumbnails(traverse_obj(
+                    block_json, ('promotionalMedia', 'crops', ..., 'renditions', ...)))
+                info = merge_dicts(content, {
+                    'title': art_title,
+                    'description': art_description,
+                    'creator': creator,
+                    'thumbnails': thumbnails,
+                })
+
+                entries.append(info)
 
-        description = track.get('description') or self._html_search_meta(
-            ['og:description', 'twitter:description'], webpage)
+            return self.playlist_result(entries, page_id, art_title, art_description)
 
-        podcast_title = audio_data.get('podcast', {}).get('title')
-        title = ('%s: %s' % (podcast_title, episode_title)
-                 if podcast_title else episode_title)
+        block_json = blocks_json.pop()
+        content = self._extract_content_from_block(block_json)
+        thumbnails = self._extract_thumbnails(traverse_obj(
+            block_json, ('promotionalMedia', 'crops', ..., 'renditions', ...)) or traverse_obj(
+                article_json, ('promotionalMedia', 'assetCrops', ..., 'renditions', ...)))
```

Comment:
this can all be inlined in the return

```py
        return merge_dicts(self._extract_content_from_block(blocks_json[0]), {
            'title': art_title,
            'description': art_description,
            'creator': creator,  # TODO: change to 'creators'
            'upload_date': unified_strdate(art_upload_date),
            'timestamp': parse_iso8601(art_upload_date),
            'thumbnails': self._extract_thumbnails(traverse_obj(
                block_json, ('promotionalMedia', 'crops', ..., 'renditions', ...)) or traverse_obj(
                article_json, ('promotionalMedia', 'assetCrops', ..., 'renditions', ...))),
        })